### DISCUSSION TITLE
Prediction: Pixel probability

### DISCUSSION THREAD

**Discussion by pauljhin** [Project-MONAI/tutorials#1270] [Q&A]
Hi,

I'm following[ spleen's tutorial](https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/spleen_segmentation_3d.ipynb) for my problem. 

I would like to get the probability of each pixel after prediction (between 0 and 1). 

I removed AsDiscreted but for each exam I get different values like for example between -10 and 12, -13 and 14 etc.

Another thing I noticed is that my segmented object takes on low (negative - black) values and background (positive - white).

See the transforms part of the prediction:

```

post_transforms = Compose(
    [
        Invertd(
            keys="pred",
            transform=test_org_transforms,
            orig_keys="image",
            meta_keys="pred_meta_dict",
            orig_meta_keys="image_meta_dict",
            meta_key_postfix="meta_dict",
            nearest_interp=False,
            to_tensor=True,
        ),
        #AsDiscreted(keys="pred", argmax=True, to_onehot=2),
        SaveImaged(keys="pred", meta_keys="pred_meta_dict", output_dir="./out", output_postfix="seg", resample=False),
    ]
)```

Any suggestion?

**Comment by pauljhin** [comment]
Rephrasing the question:

What do these prediction values that are not between 0 and 1 mean?

Why are object values low (negative) and background high (positive)?

How to get values between 0 and 1 directly, without me having to normalize?

**Comment by KumoLiu** [comment]
Hi @pauljhin, thanks for your interest here.
If you want pixel probability which between 0 and 1, you need to add `Activations` after `Invertd`.  The probabilities produced by a Sigmoid are independent. And the Softmax probabilities will always sum to one. If your output channel is 2, you could use `Activations(softmax=True)`.

Hope it can help you, thanks!

**Comment by KumoLiu** [comment]
Hi @pauljhin, thanks for your interest here.
If you want pixel probability which between 0 and 1, you need to add `Activations` after `Invertd`.  The probabilities produced by a Sigmoid are independent. And the Softmax probabilities will always sum to one. If your output channel is 2, you could use `Activations(softmax=True)`.

Hope it can help you, thanks!

**Comment by pauljhin** [comment]
Hi @KumoLiu 

Thank you very much! 

Could you help me with another question.

My code already loads the data into a numpy array and the spacing into a list. How can I inform the current spacing from my matrix for the [Spacingd](https://docs.monai.io/en/stable/transforms.html#spacingd) transformation make correct re-spacing?

UPDATE: How can I use Spacingd with just the pixel array and spacing information?

Best regards
